src/App.tsx

```diff
@@ -1,26 +1,247 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { deleteTodo, postTodo, updateTodo, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Error } from './components/Error';
+import { FilterStatus } from './types/FilterStatus';
+import { TodoList } from './components/TodoList/TodoList';
 
 export const App: React.FC = () => {
+  const focusRef = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [todosFilter, setTodosFilter] = useState(FilterStatus.All);
+  const [selectedTodo, setSelectedTodo] = useState<Todo>();
+  const [todoBeingAdded, setTodoBeingAdded] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [clearInput, setClearInput] = useState(false);
+  const [deletingIds, setDeletingIds] = useState<number[]>([]);
+  const [togglingIds, setTogglingIds] = useState<number[]>([]);
+
+  function filterTodos(filter: string) {
```

Comment:
It is important to stick to one way of creating functions, fix it everywhere
```suggestion
  const filterTodos = (filter: string) => {
```